cirq-core/cirq/study/resolver.py

```diff
@@ -39,6 +39,10 @@
 _RecursionFlag = object()
 
 
+def _is_param_resolver_or_similar_type(obj: Any):
```

Comment:
This is written here, but only used once by your code. Do you plan to have more calls coming into this ? Otherwise might be worth just getting rid of this.

Reply:
well I can't get rid of it completely since I'm using it. Would you rather I put it with the rest of the code rather than `resolver.py`?

Reply:
Fair enough, I guess we can leave it for now.